bigchaindb/common/transaction.py

```diff
@@ -612,17 +665,27 @@ def __init__(self, operation, asset, fulfillments=None, conditions=None,
         self.fulfillments = fulfillments if fulfillments else []
         self.metadata = metadata
 
+        # validate asset
+        # we know that each transaction relates to a single asset
+        # we can sum the amount of all the conditions
+
+        if self.operation == self.CREATE:
+            amount = sum([condition.amount for condition in self.conditions])
+            self.asset.validate_asset(amount=amount)
+        else:
+            # In transactions other then `CREATE` we don't know if its a
+            # divisible asset or not, so we cannot validate the amount here
+            self.asset.validate_asset()
+
     @classmethod
-    def create(cls, owners_before, owners_after, metadata=None, asset=None,
-               secret=None, time_expire=None):
+    def create(cls, owners_before, owners_after, metadata=None, asset=None):
         """A simple way to generate a `CREATE` transaction.
 
             Note:
                 This method currently supports the following Cryptoconditions
                 use cases:
                     - Ed25519
                     - ThresholdSha256
-                    - PreimageSha256.
```

Comment:
The problem with hashlocks is that its difficult to provide a generic method to create and validate them. They don't really go well with the `create, sign, validate` api.

This type of custom cryptoconditions should be added later on the driver side